pkg/components/estimator/estimator.go

```diff
@@ -0,0 +1,143 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package estimator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sustainable.computing.io/kepler-operator/pkg/api/v1alpha1"
+	"github.com/sustainable.computing.io/kepler-operator/pkg/components"
+	"github.com/sustainable.computing.io/kepler-operator/pkg/utils/k8s"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	prefix = "estimator-"
+	// exported
+	PVCName = prefix + "pvc"
+
+	StableImage          = "quay.io/sustainable_computing_io/kepler_model_server:v0.6"
+	waitForSocketCommand = "until [ -e /tmp/estimator.sock ]; do sleep 1; done && %s"
+)
+
+// Config that will be set from outside
+var (
+	Config = struct {
+		Image string
+	}{
+		Image: StableImage,
+	}
+
+	bashCommand = []string{"/bin/sh", "-c"}
+)
+
+var (
+	// common labels for all resources of modelserver
+	labels = components.CommonLabels.Merge(k8s.StringMap{
+		"app.kubernetes.io/component":  "model-server",
+		"sustainable-computing.io/app": "model-server",
+	})
+)
+
+// IsEstimatorSidecarEnabled returns true if any of estimator config has sidecar enabled
+func IsEstimatorSidecarEnabled(k *v1alpha1.Kepler) bool {
+	if k.Spec.NodeTotalEstimator != nil && k.Spec.NodeTotalEstimator.SidecarEnabled {
+		return true
+	}
+	if k.Spec.NodeComponentsEstimator != nil && k.Spec.NodeComponentsEstimator.SidecarEnabled {
+		return true
+	}
+	if k.Spec.ContainerTotalEstimator != nil && k.Spec.ContainerTotalEstimator.SidecarEnabled {
+		return true
+	}
+	if k.Spec.ContainerComponentsEstimator != nil && k.Spec.ContainerComponentsEstimator.SidecarEnabled {
+		return true
+	}
+	return false
+}
+
+// NewContainer returns sidecar container
+func NewContainer() corev1.Container {
+	mounts := []corev1.VolumeMount{{
+		Name:      "cfm",
+		MountPath: "/etc/kepler/kepler.config",
+		ReadOnly:  true,
+	}, {
+		Name:      "mnt",
+		MountPath: "/mnt",
+	}, {
+		Name:      "tmp",
+		MountPath: "/tmp",
+	}}
+
+	return corev1.Container{
+		Image:           Config.Image,
+		ImagePullPolicy: corev1.PullAlways,
+		Name:            "estimator",
+		VolumeMounts:    mounts,
+		Command:         []string{"python3.8"},
+		Args:            []string{"-u", "src/estimate/estimator.py"},
+	}
+}
+
+// NewContainer returns sidecar additional volumes
+func NewVolumes() []corev1.Volume {
+	return []corev1.Volume{
+		k8s.VolumeFromEmptyDir("mnt"),
+		k8s.VolumeFromEmptyDir("tmp"),
+	}
+}
+
+func GetModifiedVolumeMount(volumeMounts []corev1.VolumeMount) []corev1.VolumeMount {
+	return append(volumeMounts, corev1.VolumeMount{
+		Name:      "tmp",
+		MountPath: "/tmp",
+	})
+}
+
+func GetModifiedCommandAndArgs(inCmd []string) (outCmd, outArgs []string) {
+	outCmd = bashCommand
+	outArgs = []string{fmt.Sprintf(waitForSocketCommand, strings.Join(inCmd, " "))}
+	return
+}
+
+func getEstimatorConfig(prefix string, spec v1alpha1.EstimatorSpec) string {
```

Comment:
```suggestion
func estimatorConfig(prefix string, spec v1alpha1.EstimatorSpec) string {
```
Getters - https://go.dev/doc/effective_go#Getters  `GetFoobar` -> `Foobar`